dowhy/gcm/util/graphviz.py

```diff
@@ -0,0 +1,72 @@
+import os
+import tempfile
+from typing import Optional, Dict, Tuple, Any
+
+import graphviz
```

Comment:
ah, I just noticed that you are using graphviz whereas causal_graph.py uses pygraphviz.

Any preferences among the two? Would it be okay to use pygraphviz for consistency? Or if you think that graphviz is better, we can modify/integrate with the causal_graph code.

Reply:
I found ```graphviz``` better for customization and its better supported (at least based on github). I had actually problems getting the same functionality with ```pygraphviz```, but I also didn't spent too much time on it yet. In the end, both require the external graphviz installation, but of course it makes sense to only have a dependency on one of them.

Do you think we should converged there on something already or could this be a follow up? Ultimately, we should only have on plotting function in the library. For now, this just helps us with getting the existing GCM notebooks/documentation running.

Reply:
While we can keep the two different plotting files for now, it will be good to only use one of the graphviz libraries (graphviz or pygraphviz). This is to avoid confusion for the users of v0.8--will be awkward to ask them to install two plot libraries.

[pydot](https://github.com/pydot/pydot) is another contender. Its benefit is that it is pure python and networkx can directly output a pydot graph. `graph = networkx.drawing.nx_pydot.to_pydot(my_networkx_graph)`. Can you check if it is possible to write this code in pydot?

@petergtz Any thoughts here? Will be nice to converge to one library.

Reply:
Agree we should converge to one library. I wouldn't use pydot, because neither of us has used it so far.
@bloebp For the purposes we have right now, can we actually rewrite our implementation to use `pygraphviz`? I.e., does it have the features we need? My main motivation would be to avoid changing too many things at once. So unless there's a strong need for a different dependency, I would try not to change it.

Reply:
I was able to change the code to ```pygraphviz```. Was actually easier than I thought. 

Regarding the pydot plotting via networkx, the issue is that networkx can not plot curved arrows as far as I know. The graphviz plots are therefore much nicer usually.